### ISSUE TITLE
Docker db problem

### ISSUE THREAD

**Issue by lectrician1** [openstreetmap/openstreetmap-website#3312]
I followed all of the instructions in DOCKER.md and started both the web and db containers, but I get this error when first visiting the website: 
```
config/initializers/cors.rb:9:in `call'

Started GET "/" for 172.18.0.1 at 2021-09-05 15:10:06 +0000

  

ActiveRecord::NoDatabaseError (FATAL:  role "openstreetmap" does not exist

):
```

And therefore it cannot start. I just need some help to get it working.

**Comment by tomhughes** [comment]
Well unfortunately none of us knows anything about docker so you're probably not going to find much help here...

**Comment by tomhughes** [comment]
In any case there's pretty much nothing anybody can do with that tiny fragment of information - whatever the actual problem is happened long before that so we need to see the log from when the database was built and populated.

**Comment by tomhughes** [comment]
Specifically where the database container executed the `docker/postgres/openstreetmap-postgres-init.sh` script which does:

```
psql -v ON_ERROR_STOP=1 -U "$POSTGRES_USER" <<-EOSQL
    CREATE USER openstreetmap SUPERUSER PASSWORD 'openstreetmap';
    GRANT ALL PRIVILEGES ON DATABASE openstreetmap TO openstreetmap;
EOSQL
```

**Comment by lectrician1** [comment]
This first line here seems to be the problem. 
```
/usr/local/bin/docker-entrypoint.sh: /docker-entrypoint-initdb.d/openstreetmap-postgres-init.sh: /bin/bash^M: bad interpreter: No such file or directory

2021-09-05 15:04:35.412 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432

2021-09-05 15:04:35.412 UTC [1] LOG:  listening on IPv6 address "::", port 5432

2021-09-05 15:04:35.417 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"

2021-09-05 15:04:35.430 UTC [26] LOG:  database system was interrupted; last known up at 2021-09-05 14:58:13 UTC

2021-09-05 15:04:35.686 UTC [26] LOG:  database system was not properly shut down; automatic recovery in progress

2021-09-05 15:04:35.688 UTC [26] LOG:  redo starts at 0/16506A0

2021-09-05 15:04:35.688 UTC [26] LOG:  invalid record length at 0/1650790: wanted 24, got 0

2021-09-05 15:04:35.688 UTC [26] LOG:  redo done at 0/1650748

2021-09-05 15:04:35.688 UTC [26] LOG:  last completed transaction was at log time 2021-09-05 14:58:13.571741+00

2021-09-05 15:04:35.703 UTC [1] LOG:  database system is ready to accept connections

2021-09-05 15:04:43.071 UTC [33] FATAL:  role "openstreetmap" does not exist
```

I don't get how it can't find the file. It literally exists:
![image](https://user-images.githubusercontent.com/22477298/132260289-c7a0e1a6-d642-43d6-85ef-d0d6678354d7.png)

I even tried rebuilding the image and it doesn't work.

**Comment by tomhughes** [comment]
I'm not sure what you think that picture proves? It doesn't seem to show whether bash exists in the container?

**Comment by tomhughes** [comment]
As far as I can see the latest `postgres:11` container has `/bin/bash` in it so I have no idea what the problem is...

**Comment by tomhughes** [comment]
Ah the `^M` is important - you've got DOS line endings in `docker/postgres/openstreetmap-postgres-init.sh` which is what is confusing it! You need to unmangle that file to have the proper unix line endings...

**Comment by lectrician1** [comment]
That worked, thank you!

However I have one other error when first loading the site:
```
Started GET "/" for 172.22.0.1 at 2021-09-06 22:44:51 +0000

Processing by SiteController#index as HTML

  Rendering layout layouts/map.html.erb

  Rendering site/index.html.erb within layouts/map

  Rendered site/index.html.erb within layouts/map (Duration: 0.3ms | Allocations: 42)

  Rendered layout layouts/map.html.erb (Duration: 7553.7ms | Allocations: 296786)

Completed 500 Internal Server Error in 7572ms (Allocations: 299595)



ActionView::Template::Error (Error: Invalid CSS after ".": expected 1 selector or at-rule, was "../common.scss"

        on line 1:1 of app/assets/stylesheets/ltr/common.scss

>> ../common.scss


   ^

):

  
app/assets/stylesheets/ltr/common.scss:1
```

**Comment by mmd-osm** [comment]
It seems you’re using Windows, and your git has crlf auto conversion enabled by default. Can you turn that off first to preserve the original unix style line break, then delete + clone the whole repo again.

**Comment by tomhughes** [comment]
Yes my guess is that the scss file has the same line ending problem.

**Comment by gravitystorm** [comment]
@lectrician1 could you try again with a fresh checkout? Hopefully we've fixed the line-endings problem now.

**Comment by gravitystorm** [comment]
I'm going to hope that we've fixed this for you, and close this issue. But if it turns out that it's unresolved on a fresh checkout, please feel free to reopen.

**Comment by lectrician1** [comment]
I still get the same error.

This problem does not seem to be caused by line endings. 

I deleted the repository, redownloaded it, and built the image. It did not work. 

Then, I checked out the line endings for the `app/assets/stylesheets/ltr/common.scss` file and found out that **all other text files in the repository had LF line endings but this one** and `app\assets\stylesheets\ltr\small.scss`. I changed them from `CRLF` to `LF` in VSCode, deleted and rebuilt the image, and still get the error. 

So, line endings do not seem to be the problem. 

The error:
```
ActionView::Template::Error (Error: Invalid CSS after ".": expected 1 selector or at-rule, was "../common.scss"

        on line 1:1 of app/assets/stylesheets/ltr/common.scss

>> ../common.scss


   ^

):
```
seems to be an error relating to the building of the CSS. Another possible problem could be that `common.scss` and `small.css` are Symbolic Links and that could be messing with something (I don't know exactly how symbolic links work myself).
